### DISCUSSION TITLE
[A-L] (26 Gennaio 2024) Esercizio 1

### DISCUSSION THREAD

**Discussion by FeddyLix17** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#398] [Esami A-L]
Consideriamo un sistema di password formato da 2 lettere (scelte le 26 lettere dell’alfabeto latino, solo maiuscole) seguite da 3 cifre seguite da 2 caratteri speciali scelti tra $ , !, % e @ (per esempio BA344!$).

### Fila A
1. Quante password hanno la prima lettera del vostro nome come primo simbolo? [1.5 punti]

2. Quante password hanno L come prima lettera o ! come ultimo simbolo? [1.5 punti]

### Fila B
1. Quante password hanno le prime due lettere del vostro nome come primo e secondo simbolo (nello stesso ordine in cui compaiono nel vostro nome)? [1.5 punti]

2. Quante password hanno L come prima lettera o 4 come ultima cifra? [1.5 punti]

### Entrambe le file
3. Quante password contengono esattamente un 4 ed esattamente una A? [2 punti]

**Comment by FeddyLix17** [comment]
### Fila A

1. $\Large 1 * 26 * 10^3 * 4^2  = 1000 * 416 = 416.000$

2.

$$
    \Large
    \begin{align}
    (1 * 26 * 10^3 * 4^2) + (26^2 * 10^3 * 4 * 1)- (1 * 26 * 10^3 * 4 * 1) =\\
    = (1000 * 416) + (1000 * 2704) - (1000 * 104) =\\
    = 416000 + 2704000 - 104000 = 3016000
    \end{align}
$$

([principio di inclusione-esclusione](https://en.wikipedia.org/wiki/Inclusion%E2%80%93exclusion_principle) per gestire le password che hanno L come prima lettera e ! come ultimo simbolo conteggiate 2 volte)

---

### Fila B

1. $\Large 1 * 1 * 10^3 * 4^2  = 1000 * 4 = 4.000$

2.

$$
    \Large
    \begin{align}
    (1 * 26 * 10^3 * 4^2) + (26^2 * 10^2 * 1 * 4^2)- (1 * 26 * 10^2 * 4^2) =\\
    = (1000 * 416) + (100 * 10816) - (100 * 416) =\\
    = 416000 + 1081600 - 41600 = 1456000
    \end{align}
$$

([principio di inclusione-esclusione](https://en.wikipedia.org/wiki/Inclusion%E2%80%93exclusion_principle) per gestire le password che hanno L come prima lettera o 4 come ultima cifra conteggiate 2 volte)

---

### Entrambe le file

3. $\Large 2 * 25 * 3 * 9^2 * 4^2 = 194400$

**Reply by cek753** [reply]
In teoria il punto 3 dovrebbe essere (25 x2)x(9^2 x3)x4^2. Con la tua soluzione viene contato solo un caso(viene contato AL433!? ma non LA343!? o LA334!?)
